NewForce-Cohort-6/tabloidfullstack-stuffing
# Nt/14 updated create a tag

[Trello ticket 14-- Create a Tag](https://trello.com/c/bVncm1EJ/14-14-create-a-tag)

NOTE if you see another NT/14... branch, DO NOT use it because it is old and busted. **Only use `NT/14-Updated-Create-A-Tag`**

Creates:
File: TagNew.js

Changes:
Tabloid > Repositories > TagRepository.cs // AddTag(Tag tag)
Repositories > ITagRepository.cs //void AddTag(Tag tag)
Controllers > TagController.cs // public IActionResult Post(Tag tag)
Tabloid > Client > src > components > tags > TagList.js (adds a button)
Client > src > components > views >  ApplicationViews.js (adds Tag Form route)
Client > src > Managers > TagManager.js (adds POST fetch call)


Testing Instructions
1. `git add` and `git commit` whatever you're working on on your branch
2. `git checkout main`
3. `git fetch --a` (or `git fetch --all`)
4. `git checkout NT/14-Updated-Create-A-Tag` to switch to my branch.
5. If you don't see the most recent version of the files, try `git pull origin NT/14-Updated-Create-A-Tag`
6. If you haven't created the Tabloid database, please do so.
7. Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-6/tabloidfullstack-stuffing/blob/main/SQL/01_Tabloid_Create_DB.sql)
8. Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-6/tabloidfullstack-stuffing/blob/main/SQL/02_Tabloid_Seed_Data.sql)
9. In your terminal, navigate to the client folder (something like cd workspace/Tabloid/Tabloid/client)
10. If you have not run the command `npm install` in your terminal at this folder level, do so now.
11. Go back to Visual Studio. Execute the backend of Tabloid from inside of Visual Studio by clicking on the solid green arrow.
12. A browser running Swagger should pop up.
13. In your terminal, run the command `npm start`
14. A browser tab running Tabloid should appear.
15. In the browser, please log in as foo@bar.com (admin profile).
16. Use the Navbar to navigate to the Tag Management page.
17. You should see a table with one column labeled "Tags:"
18. You should see a blue button at the top of the page that says "Add new tag"
19. You should see the following tags in alphabetical order:
- C#
- Funny
- JavaScript
- Scary
20. If you have added any additional tags, they will also appear in the correct order alphabetically.
21. Click the "Add new tag" button.
22. You should be redirected to localhost:3000/tagform.
23. You should see a form labeled "Add New Tag," an empty text entry field, and a blue "Save" button.
24. Enter any tag name you wish and click "Save" button.
25. You should be redirected to the Tag Management list.
26. If the tag name you saved appears on the list in alphabetical order, the feature works.


Checklist:
[x] My code follows the style guidelines of this project
[x] I have performed a self-review of my code
[x] I have commented my code, particularly in hard-to-understand areas
[x] My changes generate no new warnings or errors
[x] I have added test instructions that prove my fix is effective or that my feature works
